File: src/levels/L24-un-solo-pago/simple.rb
# Un solo método de pago y ninguno más a la vista: el arreglo es una línea.
IVA = 0.16

class Cashier
  def initialize(barista) = @barista = barista

  # region: Cashier#charge
  def charge(order)
    total = order.price * (1 + IVA)
    CardTerminal.charge(total)
    @barista.prepare(order)
  end
  # endregion
end

# region: wiring
cashier = Cashier.new(Barista.new(Counter.new))
OrderTaker.new(cashier).take(:latte, 50)
# endregion

File: src/levels/L24-un-solo-pago/ts/simple.ts
// Un solo método de pago y ninguno más a la vista: el arreglo es una línea.
const IVA = 0.16

class Cashier {
  constructor(private barista: Barista) {}

  // region: Cashier#charge
  charge(order: Order) {
    const total = order.price * (1 + IVA)
    CardTerminal.charge(total)
    this.barista.prepare(order)
  }
  // endregion
}

// region: wiring
const cashier = new Cashier(new Barista(new Counter()))
new OrderTaker(cashier).take('latte', 50)
// endregion
